### PR TITLE
Profile LIDAR services

### PR DESCRIPTION
**Pull request by llienher**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
     "@types/jqueryui": "1.12.16",
     "@types/jsts": "0.17.9",
     "@types/node": "17.0.8",
+    "@types/offscreencanvas": "^2019.6.4",
```
Comment (sbrunner) [inline package.json:97]:
```suggestion
    "@types/offscreencanvas": "2019.6.4",
```

File: src/misc/debounce2.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// The MIT License (MIT)
```
Comment (sbrunner) [inline src/misc/debounce2.ts:1]: why not src/misc/debounce.ts ?

Comment (llienher) [reply src/misc/debounce2.ts:1]: I'm not sure but I guess the CI will probably just refuse to have both ts and js at the same time for the parser. And we still use ngeoDebounce everywhere else.

File: src/profile/d3Elevation.ts [hunk 3]
```diff
@@ -28,9 +33,34 @@ import {
...
+import {BaseType as D3SelectionBaseType} from 'd3-selection';
+import {ContainerElement as D3SelectionContainerElement} from 'd3-selection';
+import {ScaleLinear as D3ScaleLinear} from 'd3';
+import {ngeoProfileOptions as ngeoOptionsNgeoProfileOptions} from 'ngeo/options';
+import {ProfileOptions as NgeoProfileElevationComponentProfileOptions} from 'ngeo/profile/elevationComponent';
+
+type HoverCbFunction = {
+  (arg1: any, arg2: number, arg3: string, arg4: Record<any, any>, arg5: string): void;
+};
+// type Function2 = {
```
Comment (ger-benjamin) [inline src/profile/d3Elevation.ts:52]: remove me

File: src/profile/d3Elevation.ts [hunk 4]
```diff
@@ -504,19 +487,19 @@ function d3Elevation(options, functions) {
...
+      function mousemove(event: Event): void {
+        const mouseX = d3pointer(event)[0];
         const x0 = x.invert(mouseX);
 
         profile.highlight(x0);
       }
 
       /**
-       *
+       * Mouseout
```
Comment (ger-benjamin) [inline src/profile/d3Elevation.ts:500]: Well at this point, you can remove the comment.
Quick check also other comment. Remove them or better, improve them !

File: src/lidarprofile/Manager.ts [hunk 5]
```diff
@@ -32,76 +46,96 @@ import olStyleCircle from 'ol/style/Circle';
...
+  ngeoDebounce_: ngeoMiscDebounceMiscDebounce<DebounceFunction>;
 
-    /**
-     * @type {angular.IFilterService}
-     */
-    this.$filter = $filter;
+  /**
+   * @private
+   */
+  promise: undefined | Promise<never>;
```
Comment (ger-benjamin) [inline src/lidarprofile/Manager.ts:70]:
```suggestion
  promise_: undefined | Promise<never>;
```
As it's private

File: src/lidarprofile/Config.ts [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+  constructor() {
+    configuration.getConfig().subscribe({
+      next: (configuration: Configuration) => {
+        if (configuration) {
+          this.pytreeLidarprofileJsonUrl = configuration.pytreeLidarprofileJsonUrl;
+        }
+      },
+    });
+
+    //this.pytreeLidarprofileJsonUrl = pytreeLidarprofileJsonUrl;
```
Comment (ger-benjamin) [inline src/lidarprofile/Config.ts:191]: Remove me

File: src/lidarprofile/Manager.ts [hunk 7]
```diff
@@ -32,76 +46,96 @@ import olStyleCircle from 'ol/style/Circle';
...
+   * @private
+   */
+  map_: undefined | OlMap;
 
+  /**
+   * The hovered point attributes in D3 profile highlighted on the 2D map
+   */
+  cartoHighlight: OlOverlay;
+
+  lidarPointHighlight;
```
Comment (ger-benjamin) [inline src/lidarprofile/Manager.ts:88]: No type no desc ?

**Review by ger-benjamin — APPROVED** [review]
Looks working but please, take 1h to review comments, privates, coherence...

File: src/profile/d3Elevation.ts (hunk 4 above)
Comment (llienher) [reply src/profile/d3Elevation.ts:500]: I got a jsdoc error if I remove it, and a CI that fails ... Quite the same for all the useless ones in this file...

**Pull request merged**